### PR TITLE
deprecated(core): renamed various user related lib functions

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: docs/guides/actions.rst [hunk 1]
```diff
@@ -486,14 +486,14 @@ The registration action sets creates the sticky form and clears it once the acti
...
 
-   ...
-
-   $guid = register_user($username, $password, $name, $email, false, $friend_guid, $invitecode);
-
-   if ($guid) {
-      elgg_clear_sticky_form('register');
-      ....
-   }
+   elgg_register_user([
```
Comment (jeabakker) [inline docs/guides/actions.rst:489]: maybe `$user = ....`

File: engine/classes/Elgg/Users/Accounts.php [hunk 2]
```diff
@@ -163,20 +163,30 @@ public function assertValidAccountData($username, $password, $name, $email, $all
...
 	 */
-	public function register($username, $password, $name, $email, $allow_multiple_emails = false, $subtype = null, array $params = []) {
+	public function register(array $params = []) {
+		$username = (string) elgg_extract('username', $params);
+		$password = (string) elgg_extract('password', $params);
+		$name = (string) elgg_extract('name', $params);
+		$email = (string) elgg_extract('email', $params);
+		$subtype = elgg_extract('subtype', $params);
+		$language = (string) elgg_extract('language', $params, $this->translator->getCurrentLanguage());
+		$allow_multiple_emails = (bool) elgg_extract('allow_multiple_emails', $params);
```
Comment (jeabakker) [inline engine/classes/Elgg/Users/Accounts.php:188]: default `false`

File: engine/classes/ElggInstaller.php [hunk 3]
```diff
@@ -1573,32 +1573,18 @@ protected function createAdminAccount($submissionVars, $login = false) {
...
-				$submissionVars['password1'],
-				$submissionVars['displayname'],
-				$submissionVars['email']
-			);
+			$user = elgg_register_user([
+				'username' => $submissionVars['username'],
+				'password' => $submissionVars['password1'],
+				'name' => $submissionVars['displayname'],
+				'email' => $submissionVars['email'],
+			]);
```
Comment (jeabakker) [inline engine/classes/ElggInstaller.php:1581]: Why not make `$submissionVars` (eg the form) compatible with the required options an then just pass `$submissionVars`

File: engine/lib/users.php [hunk 4]
```diff
@@ -7,6 +7,7 @@
 use Elgg\Exceptions\InvalidParameterException;
 use Elgg\Exceptions\ClassException;
 use Elgg\Exceptions\Configuration\RegistrationException;
+use Alliander\Theme\User;
```
Comment (jeabakker) [inline engine/lib/users.php:10]: I think not ;)

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: engine/classes/ElggInstaller.php (hunk 3 above)
Comment (jdalsem) [reply engine/classes/ElggInstaller.php:1581]: not doing this at this moment

**Review by jeabakker — APPROVED** [review]

**Review by jeabakker — APPROVED** [review]

**Pull request merged**